.gitlab-ci.yml

```diff
@@ -95,22 +96,150 @@ whois-integration-test:
     - mariadb:10.2
   tags:
     - whois
+  stage: integration-test
+  rules:
+    - if: '$CI_COMMIT_REF_NAME == "master"'
+      when: always
+    - when: manual
+      allow_failure: true
+  script:
+    - mvn -DskipTests install
+    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -Pintegration -pl whois-api -T 1C
+
+whois-client-integration-test:
+  <<: *it_env
+  services:
+    - mariadb:10.2
+  tags:
+    - whois
+  stage: integration-test
+  rules:
+    - if: '$CI_COMMIT_REF_NAME == "master"'
+      when: always
+    - when: manual
+      allow_failure: true
+  script:
+    - mvn -DskipTests install
+    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-client -T 1
+
+
+whois-commons-integration-test:
+  <<: *it_env
+  services:
+    - mariadb:10.2
+  tags:
+    - whois
+  stage: integration-test
   rules:
     - if: '$CI_COMMIT_REF_NAME == "master"'
       when: always
     - when: manual
       allow_failure: true
   script:
     - mvn -DskipTests install
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb -Dtest.containers.disabled=true verify -PintegrationAll -pl whois-api
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-client
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-commons
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-db
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-nrtm
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-query
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-rpsl
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-scheduler
-    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-update
+    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-commons -T 1
+
+
+whois-db-integration-test:
+  <<: *it_env
+  services:
+    - mariadb:10.2
+  tags:
+    - whois
+  stage: integration-test
+  rules:
+    - if: '$CI_COMMIT_REF_NAME == "master"'
+      when: always
+    - when: manual
+      allow_failure: true
+  script:
+    - mvn -DskipTests install
+    - mvn $MAVEN_OPTS -Ddb.host=mariadb verify -PintegrationAll -pl whois-db -T 1
+
```

Comment:
Do we need to specify thread count as all other tests runs separately in different gitlab runner

Reply:
some of them are not able to work in parallel, for those it is necessary to specify this. Just query and common I think